core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -91,6 +92,51 @@ private[spark] abstract class YarnSchedulerBackend(
   }
 
   /**
+   * Override the DriverEndpoint to add extra logic for the case when an executor is disconnected.
+   * We should check the cluster manager and find if the loss of the executor was caused by YARN
+   * force killing it due to preemption.
+   */
+  private class YarnDriverEndpoint(rpcEnv: RpcEnv, sparkProperties: ArrayBuffer[(String, String)])
```

Comment:
Which other endpoint? The YarnSchedulerEndpoint? The main problem is that onDisconnected events are received at the DriverEndpoint when the executors exit from preemption. The superclass just removes the executors and marks them as SlaveLost but I want to override this.

I'll document this accordingly.